### PR TITLE
11.5.2-hotfix-2

### PR DESCRIPTION
**Pull request by krzychu124**
- Cargo Ferries mod integration - spawn cargo ferries fix,
- 81-tiles changes,
- Traffic Lights bugfix (change flags at `Untouchable` node) - compatibility with Movable Bridge mod

### CODE REVIEW

File: TLM/TLM/State/Flags.cs [hunk 1]
```diff
@@ -137,6 +137,11 @@ public class Flags {
                 return false;
             }
 
+            if ((Singleton<NetManager>.instance.m_nodes.m_buffer[nodeId].m_flags &
+                NetNode.Flags.Untouchable) != NetNode.Flags.None) {
+                return false;
+            }
+
             ItemClass connectionClass = Singleton<NetManager>
```
Comment (DaEgi01) [inline TLM/TLM/State/Flags.cs:145]: its just a minor thing but I am not sure if compiler will optimize that second call away ... I'd get the node first and refuse that in both cases.

**Review by DaEgi01 — APPROVED** [review]

**Review by DaEgi01 — APPROVED** [review]
very gud

**Review by kvakvs — APPROVED** [review]
Nice

**Review by chameleon-tbn — APPROVED** [review]
lgtm
ingame test of barges works also in heavy modded game. Side note: the Westhafen Lagerhalle https://steamcommunity.com/sharedfiles/filedetails/?id=2488858756  do not spawn barges for me. The simple barge harbour works fine with this TM:PE version

**Pull request merged**